pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -29,9 +29,17 @@ import (
 	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
 )
 
-// endpoint maps to a single registered device plugin. It is responsible
+// Endpoint maps to a single registered device plugin. It is responsible
 // for managing gRPC communications with the device plugin and caching
 // device states reported by the device plugin.
+type Endpoint interface {
```

Comment:
Any reason we need to make this a public interface? Perhaps we should call this interface endpoint and the following struct endpointImpl?

Reply:
Can make this a private one.